ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt

```diff
@@ -53,7 +53,9 @@ internal constructor(
 
   private val _uiState: MutableStateFlow<UiState> = MutableStateFlow(UiState.FetchingSurveys)
   val uiState: StateFlow<UiState> = _uiState.asStateFlow()
-  private var surveyActivationInProgress = false
+
+  private val _surveyActivationInProgress = MutableStateFlow<Boolean>(false)
+  var surveyActivationInProgress = _surveyActivationInProgress.asStateFlow()
```

Comment:
This is never actually used as a flow in this PR, since it's always being accessed via `value`. In that case, why define it as such?

Reply:
changed to primitive

Reply:
The reason to use the flow is because of baking property, without it, the value can be changed from outside, which we don't want, and if going with a primitive type, it will not trigger as soon the value changes.

Updated to primitive tye with one variable, but not a healthy approach as the value can be changed from outside which we don't want/.

Reply:
What do you mean "can be changed from outside"? The member was `private`, so the value could not be changed from outside this class.

Reply:
yes, but this PR made the variable public, so that fragment can access it

Reply:
Sorry I missed that. Can you just make it a `val` instead of a `var`?

Reply:
jumping the conversation to different thread
https://github.com/google/ground-android/pull/2768#discussion_r1782042009